File: data/data.code.lua.server/ReqChatSend.lua
module("ReqChatSend", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqChatSend})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.channel = 0
	self.dest_uid = 0
	self.content = ""
end


function decode(self, pack)
	self.channel = pack:readU8()
	self.dest_uid = pack:readU32()
	self.content = pack:readString()
	return self
end

File: examples/golang/server/proto/req_chat_send.go
package proto

import (
	"packet"
)

type ReqChatSend struct {
	Channel                  uint8
	DestUid                  uint32
	Content                  string
}

func ReqChatSendDecode(pack *packet.Packet) *ReqChatSend {
	reqChatSend := &ReqChatSend{}

	reqChatSend.Channel = pack.ReadUint8()
	reqChatSend.DestUid = pack.ReadUint32()
	reqChatSend.Content = pack.ReadString()
	return reqChatSend
}
